### PR TITLE
#855 fix and parts of new create standard color map

### PR DESCRIPTION
**Pull request by brigb123**
Edges fix for uniform colors bug. missing proper VBO uniforms and attributes. 

Also contains new widgets and features for create standard color map (still incomplete).

### CODE REVIEW

File: src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h [hunk 1]
```diff
@@ -32,6 +32,8 @@
 #include "Interface/Modules/Visualization/ui_CreateBasicColorMap.h"
 #include <Interface/Modules/Base/ModuleDialogGeneric.h>
 #include <Interface/Modules/Visualization/share.h>
+#include <Core/Datatypes/ColorMap.h>
+using namespace SCIRun::Core::Datatypes;
```
Comment (dcwhite) [inline src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h]: @brigb123 using directives in headers = no-no.

File: src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h [hunk 2]
```diff
@@ -32,6 +32,8 @@
 #include "Interface/Modules/Visualization/ui_CreateBasicColorMap.h"
 #include <Interface/Modules/Base/ModuleDialogGeneric.h>
 #include <Interface/Modules/Visualization/share.h>
```
Comment (dcwhite) [inline src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h]: Remember share.h must always be the final #include.

File: src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h [hunk 3]
```diff
@@ -46,6 +48,15 @@ class SCISHARE CreateBasicColorMapDialog : public ModuleDialogGeneric,
     SCIRun::Dataflow::Networks::ModuleStateHandle state,
     QWidget* parent = 0);
   virtual void pull();
+ private Q_SLOTS:
+  void updateColorMapPreview(QString s);
+  const QString buildGradientString(ColorMap cm);
```
Comment (dcwhite) [inline src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h]: Change to Core::Datatypes::ColorMapHandle

Comment (dcwhite) [inline src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h]: Then make corresponding changes to .cc file

File: src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h (hunk 1 above)
Comment (brigb123) [inline src/Interface/Modules/Visualization/CreateBasicColorMapDialog.h]: moved to .cc ok?

**Pull request merged**